data/storage/etcd/store_test.go

```diff
@@ -163,6 +168,77 @@ func TestUpdate(t *testing.T) {
 	}
 }
 
+func TestList(t *testing.T) {
+	// generic context
+	ctx := context.Background()
+
+	// store everything under amp/foo/
+	key := "foo"
+
+	// this is a "template" object that provides a concrete type for list to unmarshal into
+	obj := &storage.Project{}
+
+	// unlimited ttl
+	ttl := int64(0)
```

Comment:
I do get the idiom of letting Go initialize things to the [zero value](https://golang.org/ref/spec#The_zero_value), but for my tests I wanted to be explicit. I had been playing with different TTLs in the tests and decided for now to leave at 0 although I'll replace with a reasonable timeout once I'm ready to close the issue. It also matches the way I'm initializing everything else, as well as shows others how the TTL can be initialized to a different value.